tcping.go

```diff
@@ -391,6 +392,76 @@ func permuteArgs(args cliArgs) {
 	}
 }
 
+// newNetworkInterface uses the 1st ip address of the interface
+// if any err occurs it calls `tcpStats.printer.printError` and exits with statuscode 1.
+// or return `networkInterface`
+func newNetworkInterface(tcpStats *stats, netInterface string) networkInterface {
+	var interfaceAddress net.IP
+
+	// if netinterface is the addres `interfaceAddress` var will not be `nil`
+	interfaceAddress = net.ParseIP(netInterface)
+
+	if interfaceAddress == nil {
+		ief, err := net.InterfaceByName(netInterface)
+		if err != nil {
+			tcpStats.printer.printError("Interface %s not found", netInterface)
+			os.Exit(1)
+		}
+
+		addrs, err := ief.Addrs()
+		if err != nil {
+			tcpStats.printer.printError("Unable to get Interface addresses")
+			os.Exit(1)
+		}
+
+		// Iterating through the available addresses to identify valid IP configurations
+		for _, addr := range addrs {
+			if ip := addr.(*net.IPNet).IP; ip != nil {
+				// default is ipv4
+				if ip.To4() != nil && !tcpStats.userInput.useIPv6 {
+					// ipv4
+					interfaceAddress = ip
+					break
+				} else if ip.To4() == nil && tcpStats.userInput.useIPv6 {
```

Comment:
```suggestion
				} else if ip.To6() != nil && !tcpStats.userInput.useIPv4 {
```

This way we do not bug out when -4 and -6 are **not** specified

Reply:
`ip.To6` isn't a func.
and `ip.To16()` doesn't work properly

Reply:
I suggest using this newer standard package https://pkg.go.dev/net/netip for working with IPs

Reply:
thanks it's fixed now